libimage/manifests/manifests.go

```diff
@@ -263,16 +274,29 @@ func (l *list) Push(ctx context.Context, dest types.ImageReference, options Push
 		ForceCompressionFormat:           options.ForceCompressionFormat,
 	}
 
-	// Copy whatever we were asked to copy.
-	manifestBytes, err := cp.Image(ctx, policyContext, dest, src, copyOptions)
-	if err != nil {
-		return nil, "", err
+	retryOptions := retry.Options{}
+	retryOptions.MaxRetry = defaultMaxRetries
+	if options.MaxRetries != nil {
+		retryOptions.MaxRetry = int(*options.MaxRetries)
 	}
-	manifestDigest, err := manifest.Digest(manifestBytes)
-	if err != nil {
-		return nil, "", err
+	if options.RetryDelay != nil {
+		retryOptions.Delay = *options.RetryDelay
+	}
+
+	// Copy whatever we were asked to copy.
+	var manifestDigest digest.Digest
+	f := func() error {
+		opts := copyOptions
+		manifestBytes, err := cp.Image(ctx, policyContext, dest, src, opts)
+		if err == nil {
```

Comment:
Can merge these two lines, too, now that I look at it again.

Reply:
Done.